### PR TITLE
import env.STACKTOP in the wasm backend path

### PR DESCRIPTION
**Pull request by kripken**
As discussed in #1818

### CODE REVIEW

File: src/wasm/wasm-emscripten.cpp [hunk 1]
```diff
@@ -833,7 +849,9 @@ std::string EmscriptenGlueGenerator::generateEmscriptenMetadata(
   meta << "  \"externs\": [";
   commaFirst = true;
   ModuleUtils::iterImportedGlobals(wasm, [&](Global* import) {
-    meta << nextElement() << "\"_" << import->base.str << '"';
+    if (!(import->module == ENV && import->name == STACK_INIT)) {
+      meta << nextElement() << "\"_" << import->base.str << '"';
+    }
```
Comment (sbc100) [inline src/wasm/wasm-emscripten.cpp]: What happens if you don't do this?

**Comment by kripken** [comment]
Amusingly this found a bug in wasm-ctor-eval, which only handled the stack going up, but not down. After this change makes us more similar to asm2wasm, the ctor-evaller tried to do more, which hit the bug. The fix is easy though (make it not assume either direction and allow both).

File: src/wasm/wasm-emscripten.cpp (hunk 1 above)
Comment (kripken) [reply src/wasm/wasm-emscripten.cpp]: Without this it would say that there is an extern "_STACKTOP". We don't need that since the special imports are already set up. Also this would be wrong with the `_` prefix, so we'd need to fix that if we needed it (but we don't luckily).

File: src/asmjs/shared-constants.cpp [hunk 2]
```diff
@@ -67,6 +67,8 @@ cashew::IString GLOBAL("global"),
                 ALMOST_ASM("almost asm"),
                 BUFFER("buffer"),
                 ENV("env"),
+                STACKTOP("STACKTOP"),
+                STACK_MAX("STACK_MAX"),
```
Comment (sbc100) [inline src/asmjs/shared-constants.cpp]: If I was feeling picky I would ask you split this into three separate PR:

1. Moving/using of shared constants
2. Fix the ctor eval issue
3. Import STACKTOP.

Its up to you if you actaully want to do that.

Comment (kripken) [reply src/asmjs/shared-constants.cpp]: Fair point, these are separable things. I split this into 3 commits as suggested, and can land this with rebase-and-merge.
